### PR TITLE
feat: Add Geospatial query support

### PR DESCRIPTION
**Pull request by Wackymax**
| Q               | A        |
|-----------------|----------|
| Bug fix?        | no   |
| New feature?    | yes   |
| BC breaks?      | no       |
| Related Issue   | #72  |
| Need Doc update | yes   |

## Describe your change

his commit refactors the SearchState class to include new properties for location-based search parameters. The changes include adding support for aroundLatLngViaIP, aroundLatLng, aroundRadius, minimumAroundRadius, aroundPrecision, and boundingbox.

This also updates the dependencies to use ranges so that projects relying on this one don't run into inter dependency lock issues.

### CODE REVIEW

File: helper/lib/src/lib_version.dart [hunk 1]
```diff
@@ -1 +1 @@
-const libVersion = '0.4.1';
+const libVersion = '0.5.0';
```
Comment (VladislavFitz) [inline helper/lib/src/lib_version.dart:1]:
```suggestion
const libVersion = '0.4.1';
```

**Review by VladislavFitz — CHANGES_REQUESTED** [review]
Thanks for a great contribution @Wackymax !
Could you rollback all the release-related changes since we have a special release process that we run on our own. I'll keep the changelog updates you added.

File: helper/pubspec.yaml [hunk 2]
```diff
@@ -1,26 +1,26 @@
 name: algolia_helper_flutter
 description: Patterns and APIs to implement advanced search features with Algolia for Flutter
-version: 0.4.1
+version: 0.5.0
```
Comment (VladislavFitz) [inline helper/pubspec.yaml:3]:
```suggestion
version: 0.4.1
```

File: insights/pubspec.yaml [hunk 3]
```diff
@@ -1,27 +1,27 @@
 name: algolia_insights
 description: Algolia Insights for Dart provides a simple and flexible way to track events such as clicks, conversions, and views on search results.
-version: 0.2.1
+version: 0.2.2
```
Comment (VladislavFitz) [inline insights/pubspec.yaml:3]:
```suggestion
version: 0.2.1
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -128,4 +128,3 @@ Packages with other changes:
  - **FIX**: use app documents directory to store the user token if possible (#88).
  - **FIX**: Restore pub points (#75).
  - **FEAT**: Insights (#62).
-
```
Comment (VladislavFitz) [inline CHANGELOG.md:131]: Restore the trailing line here to prevent this file from showing up in the diff.

**Review by VladislavFitz — CHANGES_REQUESTED** [review]
There are some last minor fixes left, otherwise looks great.

**Review by VladislavFitz — APPROVED** [review]

**Comment by Wackymax** [comment]
@VladislavFitz regrettably this isn't working 100%. I am getting the following when specifying the bounding box params:
`"message":"Invalid value for \"insideBoundingBox\" parameter, expected float","status":400`

Seems to be related to the algolia internal libs though

**Comment by VladislavFitz** [comment]
@Wackymax the issue in the Dart client has been fixed. Could you check if it works properly on your end now?

**Comment by Wackymax** [comment]
Thank you. I will check and let you know if I continue to encounter the error.On 11 Oct 2023, at 15:48, Vladislav Fitc ***@***.***> wrote:﻿
@Wackymax the issue in the Dart client has been fixed. Could you check if it works properly on your end now?

**Pull request merged**